tck/src/test/java/org/reactivestreams/tck/PublisherVerificationTest.java

```diff
@@ -188,6 +188,51 @@ public void required_spec105_mustSignalOnCompleteWhenFiniteStreamTerminates_shou
   }
 
   @Test
+  public void optional_spec105_emptyStreamMustTerminateBySignallingOnComplete_shouldNotAllowEagerOnComplete() throws Throwable {
+    final Publisher<Integer> illegalEmptyEagerOnCompletePublisher = new Publisher<Integer>() {
+      @Override public void subscribe(final Subscriber<? super Integer> s) {
+        s.onComplete();
+      }
+    };
+
+    requireTestFailure(new ThrowingRunnable() {
+      @Override public void run() throws Throwable {
+        PublisherVerification<Integer> verification = new PublisherVerification<Integer>(newTestEnvironment(), GC_TIMEOUT_MILLIS) {
+          @Override public Publisher<Integer> createPublisher(long elements) {
+            return illegalEmptyEagerOnCompletePublisher;
+          }
+
+          @Override public long maxElementsFromPublisher() {
+            return 0;
+          }
+
+          @Override public Publisher<Integer> createErrorStatePublisher() {
+            return null;
+          }
+        };
+
+        verification.optional_spec105_emptyStreamMustTerminateBySignallingOnComplete();
+      }
+    }, "Subscriber::onComplete() called before Subscriber::onSubscribe");
+  }
+
+  @Test
+  public void optional_spec105_emptyStreamMustTerminateBySignallingOnComplete_shouldBeSkippedForNonEmptyPublisher() throws Throwable {
+    final Publisher<Integer> failingPublisher = new Publisher<Integer>() {
+      @Override public void subscribe(final Subscriber<? super Integer> s) {
+        throw new TestException();
+      }
+    };
+
+    requireTestSkip(new ThrowingRunnable() {
+      @Override
+      public void run() throws Throwable {
+        customPublisherVerification(failingPublisher).optional_spec105_emptyStreamMustTerminateBySignallingOnComplete();
+      }
+    }, "This test is specific for 0-length streams");
```

Comment:
Perhaps I'm daft today, but I don't understand this test case. Why does it throw an exception in subscribe?